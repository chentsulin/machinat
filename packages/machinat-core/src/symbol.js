// @flow
export const MACHINAT_ELEMENT_TYPE = Symbol('machinat.element');

export const MACHINAT_FRAGMENT_TYPE = Symbol('machinat.element.fragment');

export const MACHINAT_PAUSE_TYPE = Symbol('machinat.element.pause');

export const MACHINAT_PROVIDER_TYPE = Symbol('machinat.element.provider');

export const MACHINAT_THUNK_TYPE = Symbol('machinat.element.thunk');

export const MACHINAT_RAW_TYPE = Symbol('machinat.element.raw');

export const MACHINAT_NATIVE_TYPE = Symbol('machinat.component.native');

export const MACHINAT_SERVICES_CONTAINER = Symbol(
  'machinat.services.container'
);

export const MACHINAT_SERVICES_INTERFACE = Symbol(
  'machinat.services.interface'
);

export const MACHINAT_SERVICES_PROVIDER = Symbol('machinat.services.provider');
