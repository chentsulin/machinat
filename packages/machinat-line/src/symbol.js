export const LINE_NAITVE_TYPE = Symbol.for('machinat.line.native');

export const API_PATH = Symbol.for('machinat.line.api_path');
export const HAS_BODY = Symbol.for('machinat.line.has_body');
export const THREAD_ID = Symbol.for('machinat.line.thread_id');

export const NO_BODY = Symbol.for('machinat.line.act.no_body');
