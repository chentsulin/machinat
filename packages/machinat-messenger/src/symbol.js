export const MESSENGER_NAITVE_TYPE = Symbol.for('machinat.messenger.native');
