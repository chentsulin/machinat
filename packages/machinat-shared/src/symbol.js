export const MACHINAT_ELEMENT_TYPE = Symbol.for('machinat.element');

export const MACHINAT_ROOT_ELEMENT = Symbol.for('machinat.element.root');

export const MACHINAT_FRAGMENT_TYPE = Symbol.for('machinat.element.fragment');

export const MACHINAT_ASYNC_TYPE = Symbol.for('machinat.element.async');
