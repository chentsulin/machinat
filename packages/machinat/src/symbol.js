// @flow
export const MACHINAT_ELEMENT_TYPE = Symbol.for('machinat.element');

export const MACHINAT_FRAGMENT_TYPE = Symbol.for('machinat.element.fragment');

export const MACHINAT_PAUSE_TYPE = Symbol.for('machinat.element.pause');

export const MACHINAT_NATIVE_TYPE = Symbol.for('machinat.component.native');

export const MACHINAT_SERVICE_TYPE = Symbol.for('machinat.service');

export const MACHINAT_PROVIDER_TYPE = Symbol.for('machinat.service.provider');

export const MACHINAT_CONSUMER_TYPE = Symbol.for('machinat.service.consumer');

export const SEGMENT_BREAK = Symbol.for('machinat.segment.break');
