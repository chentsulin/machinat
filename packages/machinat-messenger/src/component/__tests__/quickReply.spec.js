import Machinat from 'machinat';

import { MESSENGER_NAITVE_TYPE } from '../../symbol';
import {
  QuickReply,
  PhoneQuickReply,
  EmailQuickReply,
  LocationQuickReply,
} from '../quickReply';
import renderHelper from './renderHelper';

const render = renderHelper(null);

it.each([QuickReply, PhoneQuickReply, EmailQuickReply, LocationQuickReply])(
  '%p is valid Component',
  Reply => {
    expect(typeof Reply).toBe('function');
    expect(Reply.$$native).toBe(MESSENGER_NAITVE_TYPE);
    expect(Reply.$$entry).toBe(undefined);
    expect(Reply.$$unit).toBe(false);
  }
);

it.each([
  <QuickReply title="i want a pie" payload="🥧" />,
  <QuickReply title="a piece of cake" payload="🍰" imageURL="http://cake.it" />,
  <PhoneQuickReply />,
  <EmailQuickReply />,
  <LocationQuickReply />,
])('%p match snapshot', element => {
  expect(render(element)).toMatchSnapshot();
});
