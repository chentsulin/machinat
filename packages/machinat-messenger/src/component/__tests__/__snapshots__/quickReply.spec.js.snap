// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`{"$typeof": Symbol(machinat.element), "props": [Object], "type": [Function EmailQuickReply]} match snapshot 1`] = `
Array [
  Object {
    "content_type": "user_email",
  },
]
`;

exports[`{"$typeof": Symbol(machinat.element), "props": [Object], "type": [Function LocationQuickReply]} match snapshot 1`] = `
Array [
  Object {
    "content_type": "location",
  },
]
`;

exports[`{"$typeof": Symbol(machinat.element), "props": [Object], "type": [Function PhoneQuickReply]} match snapshot 1`] = `
Array [
  Object {
    "content_type": "user_phone_number",
  },
]
`;

exports[`{"$typeof": Symbol(machinat.element), "props": [Object], "type": [Function QuickReply]} match snapshot 1`] = `
Array [
  Object {
    "content_type": "text",
    "image_url": undefined,
    "payload": "ü•ß",
    "title": "i want a pie",
  },
]
`;

exports[`{"$typeof": Symbol(machinat.element), "props": [Object], "type": [Function QuickReply]} match snapshot 2`] = `
Array [
  Object {
    "content_type": "text",
    "image_url": "http://cake.it",
    "payload": "üç∞",
    "title": "a piece of cake",
  },
]
`;
